src/differentials.jl

```diff
@@ -191,6 +208,14 @@ end
     return element, (externed, new_state)
 end
 
+unthunk(x) = x
+unthunk(x::AbstractThunk) = unthunk(x())
```

Comment:
It is a little uncomfortable having `x()`, `unthunk(x)` and `extern(x)`

Reply:
We could do without `x()`, if that's bothering you. Don't know whether this will really be a problem though.